google/resumable_media/_upload.py

```diff
@@ -807,6 +808,13 @@ def get_next_chunk(stream, chunk_size, total_bytes):
         if num_bytes_read < chunk_size:
             # We now **KNOW** the total number of bytes.
             total_bytes = end_byte + 1
+    elif total_bytes == 0:
+        # NOTE: We also expect ``start_byte == 0`` here but don't check
+        #       because ``_prepare_initiate_request()`` requires the
+        #       stream to be at the beginning.
+        if num_bytes_read != 0:
+            raise ValueError(
+                u'Stream specified as empty, but produced non-empty content.')
```

Comment:
@lukesneeringer PTAL.

In either the `total_bytes == 0` and `total_bytes is None` case, we end up with:

```python
>>> get_content_range(0, -1, 0)
u'bytes */0'
```

If we "corrected" the value, we'd instead have:

```python
>>> get_content_range(0, 0, 0)
u'bytes 0-0/0'
```

IMO it's not worth correcting, LMK what you think